src/python/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+find_package(Cython REQUIRED)
+find_package(NumPy REQUIRED)
+
+add_cython_target(cy_pgeon pgeon.pyx)
+add_library(cy_pgeon MODULE ${cy_pgeon})
+set_target_properties(cy_pgeon PROPERTIES PREFIX "" OUTPUT_NAME "_pgeon")
+
+target_link_libraries(cy_pgeon PRIVATE pgeon arrow_shared ${PYTHON_LIBRARY})
```

Comment:
I think we are missing `${ARROW_PYTHON_SHARED_LIB}`
See `https://github.com/apache/arrow/blob/master/cpp/cmake_modules/FindArrowPython.cmake`

Reply:
I hardcoded the `ArrowPython` path in the new revision I just pushed, and the `_pgeon.so` is correctly linked against `arrowpython`: 
```
~/git/pgeon/dist/pgeon]$ ldd _pgeon.so 
        linux-vdso.so.1 (0x00007ffdcb7fb000)
        libarrow.so.700 => /home/rdibattista/mambaforge/envs/pgeon-dev/lib/libarrow.so.700 (0x000075470467c000)
        libarrow_python.so.700 => /home/rdibattista/mambaforge/envs/pgeon-dev/lib/libarrow_python.so.700 (0x0000754705f95000)
```

but I still get the problem with the missing symbols: 
```
~/git/pgeon/dist/pgeon]$ python -c "from pgeon import *"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/rdibattista/mambaforge/envs/pgeon-dev/lib/python3.10/site-packages/pgeon/__init__.py", line 3, in <module>
    from ._pgeon import copy_query
ImportError: /home/rdibattista/mambaforge/envs/pgeon-dev/lib/python3.10/site-packages/pgeon/_pgeon.so: undefined symbol: _ZNK5arrow8DataType18ComputeFingerprintB5cxx11Ev
```

Some investigation: 
```
find /home/rdibattista/mambaforge/envs/pgeon-dev/lib -type f -name "*.so" -exec sh -c  "objdump -T {}  | grep  '_ZNK5arrow8DataType18ComputeFingerprint'" \;
0000000000000000      DF *UND*  0000000000000000              _ZNK5arrow8DataType18ComputeFingerprintEv
0000000000000000      DF *UND*  0000000000000000              _ZNK5arrow8DataType18ComputeFingerprintB5cxx11E
```

That `arrow8` in the string makes me think it's somehow getting `arrow8` and not arrow7`

Reply:
I think the solution is to link against the python wheel stuff only when building on Python. I'll write a custom `findArrow.cmake` module.